tests/manager/scripts/test_playtime_entries.py

```diff
@@ -681,37 +680,46 @@ def test_do_run(
         )
 
         reporting_name = "test cm"
+        parent_folder_id = "palace-test"
+
+        output_data: dict[str, str] = {}
+
+        def mock_create_file(
+            file_name: str,
+            stream: IOBase,
+            content_type: str,
+            parent_folder_id: str | None = None,
+        ) -> dict[str, Any]:
+            nonlocal output_data
+            stream.seek(0)
+            output_data[file_name] = stream.read().decode("utf-8")
+            return DEFAULT
 
         mock_google_drive_service = create_autospec(GoogleDriveService)
-        services_fixture.services.google_drive.service.override(
-            mock_google_drive_service
+        mock_google_drive_service.create_file.side_effect = mock_create_file
+        drive_container = services_fixture.services.google_drive()
+        drive_container.config.from_dict({"parent_folder_id": parent_folder_id})
+        drive_container.service.override(mock_google_drive_service)
+        monkeypatch.setenv(
+            Configuration.REPORTING_NAME_ENVIRONMENT_VARIABLE, reporting_name
         )
 
-        parent_folder_id = "palace-test"
-
-        with (
-            patch("palace.manager.scripts.playtime_entries.csv.writer") as writer,
-            patch(
-                "palace.manager.scripts.playtime_entries.os.environ",
-                new={
-                    Configuration.REPORTING_NAME_ENVIRONMENT_VARIABLE: reporting_name,
-                    PALACE_GOOGLE_DRIVE_PARENT_FOLDER_ID_ENVIRONMENT_VARIABLE: parent_folder_id,
-                },
-            ),
-        ):
-            # Act
-            PlaytimeEntriesReportsScript(db.session).run()
```

Comment:
I updated this so that we're not patching the csv writer. That way we can test that the data makes it all the way into the stream.